Make doors autolock when broken, add ability to break door bolts

Soadreqm opened the pull request:
~~Make all doors bolt themselves when broken, and add the ability to break door bolts. This makes door bolts less robust, but hopefully in a way that makes sense. I was aiming for the general ballpark of bolted doors being harder to dismantle than walls, but easier than r-walls. It took me just over a minute to break through a door with equipment available to everyone - a fire extinguisher to smash it, and a welding torch to cut it open afterwards. Other usable tools are plasma cutters, energy swords and circular saws.~~

~~The fire axe also gets a boost. It can smash the door electronics in a single swing, and bypass the first step of the two-step bolt removal process, giving it some more plausible deniability as an entry tool.~~

Do not make all doors bolt themselves when broken - just modify the process of dealing with bolted, broken doors. The ones with secure_wires bolt, as they did before.

Fire axe is buffed, but still requires multiple clicks to get through a door.

Soadreqm committed:
Made doors autolock when broken, added ability to break door bolts

Soadreqm committed:
Adjusted time, added changelog

Soadreqm committed:
fix silly mistakes

PsiOmegaDelta commented on code/game/machinery/doors/airlock.dm:
```diff
@@ -14,6 +14,7 @@
 	var/spawnPowerRestoreRunning = 0
 	var/welded = null
 	var/locked = 0
+	var/lock_cut_state = 1	// Not a boolean. If 1, door bolts are fine. If 2, the bolt cover is off. If 3, the bolts have been cut.
```
Then define flags or other named values.

PsiOmegaDelta commented on html/changelogs/example.yml:
```diff
@@ -33,5 +33,7 @@ delete-after: True
+  - rscadd: "Made all doors bolt themselves when broken."
+  - rscadd: "Made it possible to cut the bolts of a broken door."
+  - rscadd: "As a side effect, it is possible to remove bolted, broken doors without an RCD."
+  - rscadd: "Also made the fire axe much more robust against doors."
```
Prepare for Travis to fail the build due to having changed example.yml.

Soadreqm replied:
Sweet Nar-Sie what are those things

PsiOmegaDelta replied:
^M? In another context it means Ctrl+M.

PJB3005 replied:
Looking at a Hex editor i see nothing wrong except a CRLF tab where the `^M` is.

Strange.

![](http://i.imgur.com/PrH8PHQ.png)

Soadreqm replied:
My best guess is it's a line ending thing.
Anyway, I put the changelog file where it's supposed to be.

PsiOmegaDelta commented on code/game/machinery/doors/airlock.dm:
```diff
@@ -778,6 +779,68 @@ About the new airlock wires panel:
+	else if(istype(item,/obj/item/weapon/material/twohanded/fireaxe))
+		//special case - zero delay, different message
+		if (src.lock_cut_state == 2)
+			..()
```
There is no base proc to call, as you're already in the base definition.
Did you mean .()?

babydoll commented:
I think this warrants some discussion on The Board

Raptor1628 commented:
Yeah, we already have the feature that some doors bolt when locked, it's just a variable in airlocks. Bolts can already be disengaged, it just takes some time. The current system of being able to break open some doors without complex tools works, this only means that you can't get through ANY airlock without tools, so why ever break one open?

Soadreqm commented:
The current behaviour is that you can get through almost any door with nothing but a crowbar and a fire extinguisher, since the secure airlock electronics are used basically nowhere. The vault and maybe the AI upload? Also, if someone does manage to break a bolted door, it can only be removed with an RCD. It feels weird.

But yeah. I was hoping to just sneak this in without anyone noticing, but I guess I'll make a forum thread. 😄 
https://baystation12.net/forums/threads/discussion-feedback-door-bolts.3336/

Raptor1628 commented:
The Torch has more than a dozen secure airlocks, I was thinking of making all command airlocks secure in their actual type.

mwerezak commented on code/game/machinery/doors/airlock.dm:
```diff
@@ -906,8 +990,7 @@ About the new airlock wires panel:
 	stat |= BROKEN
-	if (secured_wires)
-		lock()
+	lock()
```
I don't understand why you did this. The entire point of the `secured_wires` variable is to make the bolts drop when the door breaks.

mwerezak replied:
I guess there's the extra random wires thing too, but this behaviour should probably be reserved for secure areas that really need it anyways.

mwerezak commented:
I'm sort of puzzled about why you didn't just use the existing `secured_wires` var.

On the other hand I'm glad to see an implementation of something to allow players to deal with broken doors that have been bolted, since the current solution of "weld it, use crowbar, then dismantle" was really more of a bandaid solution (and perhaps a little too quick to perform). Make sure you remove that if this goes through.

Soadreqm commented:
What I'm getting from the thread is that people are generally in favor of keeping the majority of doors flimsy. Oh, well. I guess I'll put the secured_wires check back in.

Without the all-doors-autobolt bit, the changes amount to making the deconstruction process for bolted, broken doors - which IS possible, as it turns out - take a few more steps. And the fire axe one-shotting doors, which I think I should nerf a bit if most doors are to keep not bolting themselves.

Soadreqm committed:
Restore check for secured wires - most doors no longer bolt when broken
Adjust welding delay again
Disable old method of dismantling broken, bolted doors
Make fire axe power smash only work on doors that are already damaged
Modify changelog

mwerezak commented:
I still think that adding some proper steps for the deconstruction process of bolted, broken doors is a valuable addition, so thanks for that.

mwerezak commented:
@PsiOmegaDelta just realized I probably should have left it for you to decide whether your points had been satisfied. I haven't heard anything about if there is a procedure about that, but it would have been more polite, sorry.

Hubblenaut commented:
Yes, the whole point of breaking airlocks with force was to add an alternate way of opening them without extense IC knowledge of using tools.
It is also what the fire axe is specifically designed to do, opposed to being taken for self defense the first moment an antagonist reveals himself somewhere on the station.

Raptor1628 commented:
~~Fire axes don't one-shot doors, they're supposed to be effective at gaining access to areas quickly, that's their entire point.~~ Never mind you have your own changes to axes.
